src/shared/psi-util.c

```diff
@@ -0,0 +1,73 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "fileio.h"
+#include "psi-util.h"
+#include "stat-util.h"
+
+int read_resource_pressure(const char *path, PressureType type, ResourcePressure *ret) {
+        _cleanup_free_ char *value = NULL;
+        float avg10, avg60, avg300;
+        const char *field;
+        char *marker;
+        usec_t total;
+        char buf[6];
+        int r;
+
+        assert(path);
+        assert((type == PRESSURE_TYPE_SOME) || (type == PRESSURE_TYPE_FULL));
+        assert(ret);
+
+        if (type == PRESSURE_TYPE_SOME)
+                field = "some";
+        else if (type == PRESSURE_TYPE_FULL)
+                field = "full";
+
+        r = read_full_virtual_file(path, &value, NULL);
+        if (r < 0)
+                return r;
+
+        marker = value;
+
+        do {
+                r = sscanf(marker, "%5s avg10=%f avg60=%f avg300=%f total="USEC_FMT, buf, &avg10, &avg60, &avg300, &total);
+                if (r != 5)
+                        return -EINVAL;
+                else if (strcmp(buf, field) == 0)
+                        break;
+
+                marker = strstr(value, "\n");
+        } while (marker);
```

Comment:
hmm, so the file is first read in full, but then processed line-by-line. Maybe just use read_line() here, and actually read it line by line, leaving stdio to do its thing with buffer the whole thing?

Reply:
I feel like I used read_line() in an old iteration and ran into something but I don't remember what. I'll see what I come up with using `extract_first_word()`